jetty-ee9/jetty-ee9-webapp/src/main/java/org/eclipse/jetty/ee9/webapp/WebAppContext.java

```diff
@@ -1504,6 +1543,6 @@ private static void addClasses(ClassMatcher matcher, String attribute, Attribute
         int l = classes.length;
         classes = Arrays.copyOf(classes, l + pattern.length);
         System.arraycopy(pattern, 0, classes, l, pattern.length);
-        attributes.setAttribute(attribute, classes);
+        ServletContextHandler.ENVIRONMENT.setAttribute(attributeKey, classes);
```

Comment:
Why cannot system and server classes be a non-static field in `WebAppContext`?

The attributes are only used internally by `WebAppContext`, so there is no need to store them into a `static` constant, which will cause troubles in case of multiple `Server` instances in the same JVM.

Reply:
This method is used for configuring all WebAppContext's that get created in a common way.
There is already WebAppContext instance specific ways of configuring the ClassMatcher used by the WebApp.
Take a look at `WebAppContext.setServer(Server)` for where the instance specific ClassMatcher is configured.